rdm/util.py

```diff
@@ -66,13 +68,20 @@ def copy_directory(dir_source, dir_dest):
 def context_from_data_files(data_filenames):
     context = {}
     for data_filename in data_filenames:
-        key, _ = os.path.splitext(os.path.basename(data_filename))
+        key, ext = os.path.splitext(os.path.basename(data_filename))
         if key in context:
             raise ValueError('There is already data attached to the key "{}"'.format(key))
         with open(data_filename, 'r') as data_file:
             data_string = data_file.read()
         try:
-            data = yaml.load(data_string)
+            if ext == ".xml":
+                data = xmltodict.parse(data_string)
```

Comment:
I don't think that `context_from_data_files` should be aware of other file formats.

Instead, I think we should:

1. Convert test-platform-specific xml to a generic text-result yaml format
2. Load this generic yaml format as context

I think there are a number of advantages to this approach, but we can discuss.

Reply:
Notes from our meeting:

1. Add a new `rdm tests` subcommand.  This will take a `--format` commandline argument which will pick between a few different backends, similar to how `rdm pull` picks different backends, only `rdm pull` decides which backend to use based off of `system.yml`.  In the case of tests it seems like a commandline argument is more appropriate.
2. The intermediate YAML format should be very simple and generic.  We should update `rdm doctor` to handle this new format.  I think the only keys we need are: name (string), result (string, enum---pass/fail---we can probably just exclude skipped tests completely when pulling in from the backend), requirements (list of strings--we use strings for all our ids since some ids are not numbers), notes (optional; string)
3. We should update the `init_files` to include a basic test_record.md document; we should also add a basic `data/manual_tests.yml` file here, since I think most projects will have some manual tests.  We should also add appropriate TODOs in the `init_files/documents/test_record.yml`.

Reply:
"tests" has too many meanings. I used "translate" as the subcommand.

the formats are named `gtest` and `qttest` to match the include file names

I kept the 'skip' case. It helps to distinguish between having a requirement for a test that does not exist, or was deleted, versus skipping the test. The report itself can suppress those tests if they interfere, or can add a note explaining the skip.

Reply:
Thanks! I will probably not get to review this until next Monday, but it is on my TODO list.  I look forward to reading through the code.

Reply:
I agree that "tests" is overloaded, but "translate" seems generic.  I.e., converting "markdown to tex" can be seen as translating something.  I think it is fine for now, but I wouldn't be surprised if we rename our commands before we release v1.0 (the "pull" command is also poorly named).

Have we addressed bullets 2 and 3 in my first list?